src/otaclient/grpc/api_v2/servicer.py

```diff
@@ -111,6 +112,22 @@ async def update(
         update_acked_ecus = set()
         response = api_types.UpdateResponse()
 
+        # NOTE(20241220): due to the fact that OTA Service API doesn't have field
+        #                 in UpdateResponseEcu msg, the only way to pass the failure_msg
+        #                 to upper is by status API.
+        if not otaclient_cfg.ECU_INFO_LOADED_SUCCESSFULLY:
+            logger.error(
+                "ecu_info.yaml is not loaded properly, reject any update request"
+            )
+            for _update_req in request.iter_ecu():
+                response.add_ecu(
+                    api_types.UpdateResponseEcu(
+                        ecu_id=_update_req.ecu_id,
+                        result=api_types.FailureType.UNRECOVERABLE,
```

Comment:
Just comment: In future, I would like to extend these error types to share much more precise reason/playbook.

Reply:
Yes, I also want to do that! :crying_cat_face: 

I just have no idea why the designer of the API doesn't include a field for including failure_msg. Might he thought that the status API can do the job of providing the failure_msg? :face_with_head_bandage: But the failure message of rejecting a request is different from the failure message of the otaclient itself.